### ISSUE TITLE
Dialogic crashes exports

### ISSUE THREAD

**Issue by NILL2021** [coppolaemilio/dialogic#898]
## The problem

**Describe the bug**
When running an exported version of my game, dialogic crashes it with this error message:
````
SCRIPT ERROR: Invalid call. Nonexistent function 'get_value' in base 'Nil'.
          at: _process (res://addons/dialogic/Nodes/DialogNode.gdc:424)
````
This doesn't happen when running from the editor.

**To Reproduce**
I don't know how I did it so I can't show you how.

**Expected behavior**
I expect dialogue to show up when I start my game

**Screenshots**
![Screenshot 2022-05-16 204614](https://user-images.githubusercontent.com/81653971/168670498-290c9284-604d-4282-be6e-d1f226c4f656.png)

**System (please complete the following information):**
 - OS: Windows 11
 - Godot Version: 3.4.3
 - Dialogic Version: 1.4.1

## Solutions
I don't know

**Workaround**

A delay before adding dialogic as a child, this didn't work

**Possible fixes**
I have no idea :/

**Comment by NILL2021** [comment]
> Is this an export to android by chance? Either way I believe this is a known issue, and the fix is currently in main branch. Try downloading that and see if the error persists.

Okay thanks, and no,this is a windows export

**Comment by NILL2021** [comment]
> Is this an export to android by chance? Either way I believe this is a known issue, and the fix is currently in main branch. Try downloading that and see if the error persists.

Okay thanks, and no,this is a windows export

**Comment by NILL2021** [comment]
I tried that, but the problem is still there. I no get this errors:
````
**SCRIPT ERROR: Invalid call. Nonexistent function 'get_value' in base 'Nil'.
          at: parse_alignment (res://addons/dialogic/Parser/DialogicParser.gdc:178)
SCRIPT ERROR: Invalid type in function 'parse_definitions' in base 'GDScript'. Cannot convert argument 2 from Nil to String.
          at: update_text (res://addons/dialogic/Nodes/DialogNode.gdc:1036)
SCRIPT ERROR: Invalid get index 'waiting' (on base: 'Dictionary').
          at: play_audio (res://addons/dialogic/Nodes/DialogNode.gdc:392)
**
````

**Comment by matiwierzba** [comment]
Repro (I guess it's the same issue?) on Windows 10, Godot 3.4.2, Dialogic 1.4.1:

> [D] An error occurred when trying to access the custom event folder.
> SCRIPT ERROR: Invalid call. Nonexistent function 'get_value' in base 'Nil'.
>           at: resize_main (res://addons/dialogic/Nodes/DialogNode.gdc:210)
> SCRIPT ERROR: Invalid call. Nonexistent function 'get_children' in base 'Nil'.
>           at: clear_options (res://addons/dialogic/Nodes/DialogNode.gdc:1062)
> SCRIPT ERROR: Invalid call. Nonexistent function 'get_value' in base 'Nil'.
>           at: fade_in_dialog (res://addons/dialogic/Nodes/DialogNode.gdc:1386)
> SCRIPT ERROR: Invalid call. Nonexistent function 'get_value' in base 'Nil'.
>           at: update_name (res://addons/dialogic/Nodes/DialogNode.gdc:1015)

I tried exporting the game both to HTML and Windows. It crashes as soon as I try instantiate any dialog with Dialogic.

**Comment by zaknafean** [comment]
> Repro (I guess it's the same issue?) on Windows 10, Godot 3.4.2, Dialogic 1.4.1:
> 
> > [D] An error occurred when trying to access the custom event folder.
> > SCRIPT ERROR: Invalid call. Nonexistent function 'get_value' in base 'Nil'.
> > at: resize_main (res://addons/dialogic/Nodes/DialogNode.gdc:210)
> > SCRIPT ERROR: Invalid call. Nonexistent function 'get_children' in base 'Nil'.
> > at: clear_options (res://addons/dialogic/Nodes/DialogNode.gdc:1062)
> > SCRIPT ERROR: Invalid call. Nonexistent function 'get_value' in base 'Nil'.
> > at: fade_in_dialog (res://addons/dialogic/Nodes/DialogNode.gdc:1386)
> > SCRIPT ERROR: Invalid call. Nonexistent function 'get_value' in base 'Nil'.
> > at: update_name (res://addons/dialogic/Nodes/DialogNode.gdc:1015)
> 
> I tried exporting the game both to HTML and Windows. It crashes as soon as I try instantiate any dialog with Dialogic.

Can you try using the main branch from git hub (our 1.4.2) release candidate, and see if you get the same issue?

**Comment by matiwierzba** [comment]
Thanks, I just tried that! Now the game doesn't crash, but I get this error:

> [D] An error occurred when trying to access the custom event folder.
> SCRIPT ERROR: Invalid call. Nonexistent function 'get_value' in base 'Nil'.
>           at: resize_main (res://addons/dialogic/Nodes/DialogNode.gdc:214)
> SCRIPT ERROR: Invalid call. Nonexistent function 'get_children' in base 'Nil'.
>           at: clear_options (res://addons/dialogic/Nodes/DialogNode.gdc:1073)
> SCRIPT ERROR: Invalid call. Nonexistent function 'get_value' in base 'Nil'.
>           at: fade_in_dialog (res://addons/dialogic/Nodes/DialogNode.gdc:1400)

The dialog spawns, but it seems that at first it's invisible - I need to press an action button, and only then it fades in, simultaneously skipping from the first line to the second one.

**Comment by zaknafean** [comment]
> 

Can you screen shot the start of the timeline in question if you have a moment?

**Comment by matiwierzba** [comment]
I tried it with two different timelines with the same result, here they are:

![obraz](https://user-images.githubusercontent.com/61792438/170112177-7c019544-88ec-402f-bd26-2ee4455f005b.png)
![obraz](https://user-images.githubusercontent.com/61792438/170112305-89e546ac-c981-4db9-9179-2f025fba2b76.png)

**Comment by matiwierzba** [comment]
That's right

**Comment by zaknafean** [comment]
I have a hunch... are both those characters using the same theme? Or are you only using the default theme? Or did you edit the default theme and are using the edited theme for those characters?

**Comment by matiwierzba** [comment]
Yeah, the last one. I edited the default theme. Using a completely new theme should solve it?

**Comment by zaknafean** [comment]
I'm thinking it might. We get some WEIRD issues with the default theme. Give it a try and report back if you wouldn't mind.

**Comment by matiwierzba** [comment]
Still the same - I created a new theme and assigned it to both characters, but I'm getting the same error as previously and I still need to press a button to make the dialog actually appear.

**Comment by zaknafean** [comment]
More for my own notes than anything else as this isn't actionable...

It appears that the current_theme isn't being set on time. I can't currently recreate the issue, but its clearly affecting a bunch of our users.

@matiwierzba Is there any further stack trace you could share on the issue, or is that it?

**Comment by matiwierzba** [comment]
Hello again! I stumbled upon a solution for this problem, maybe it will work for others as well. In Dialogic settings I had the "default theme" still set as the default theme (I know how it sounds but I think you know what I mean). None of my characters were using it, but after I changed the default theme in the settings to an entirely new theme the issues disappeared completely, both on Windows and in the web version

**Comment by kakurokan** [comment]
> Hello again! I stumbled upon a solution for this problem, maybe it will work for others as well. In Dialogic settings I had the "default theme" still set as the default theme (I know how it sounds but I think you know what I mean). None of my characters were using it, but after I changed the default theme in the settings to an entirely new theme the issues disappeared completely, both on Windows and in the web version

This saved my life, thank you.

**Comment by coppolaemilio** [comment]
Is this closeable?

**Comment by zaknafean** [comment]
Gotta make the actual fix. Open for now will have it worked out this week.

**Comment by zaknafean** [comment]
This and its dupes should be fixed with #978
